### PR TITLE
Flapjack to Master

### PR DESCRIPTION
**Pull request by kurtw**
This merges the flapjack branch into master.  @anselmbradford @contolini are these commits good to go?

### CODE REVIEW

File: cfgov/v1/signals.py [hunk 1]
```diff
@@ -52,6 +53,12 @@ def unshare_all_revisions(sender, **kwargs):
 def unpublish_all_revisions(sender, **kwargs):
     update_all_revisions(kwargs['instance'], 'live')
 
+def configure_page_and_revision(sender, **kwargs):
+    from .wagtail_hooks import share, configure_page_revision, flush_akamai
```
Comment (vccabral) [inline cfgov/v1/signals.py]: do you have to do a line import?

File: cfgov/v1/tests/test_wagtail_hooks.py [hunk 2]
```diff
@@ -179,17 +185,36 @@ def test_save_a_draft(self):
...
         assert latest_content['live']
+
+
+class TestFlushAkamai(TestCase):
+
+    def setUp(self):
+        self.page = mock.Mock()
+        self.page.owner.email = 'test@mail.com'
+
+    @mock.patch('v1.wagtail_hooks.settings')
```
Comment (vccabral) [inline cfgov/v1/tests/test_wagtail_hooks.py]: 👍 nice use of patch

File: cfgov/v1/wagtail_hooks.py [hunk 3]
```diff
@@ -22,8 +23,14 @@ def share_the_page(request, page):
     is_sharing = bool(request.POST.get('action-share', False))
 
     check_permissions(parent_page, request.user, is_publishing, is_sharing)
-    share(page, is_sharing, is_publishing)
-    configure_page_revision(page, is_sharing, is_publishing)
+
+    is_live = False
```
Comment (vccabral) [inline cfgov/v1/wagtail_hooks.py]: these CAN be 1 one liner and a function.

File: cfgov/v1/wagtail_hooks.py [hunk 4]
```diff
@@ -33,37 +40,42 @@ def check_permissions(parent, user, is_publishing, is_sharing):
...
-            if url_paths[0] == '/':
-                is_home_page = True
-            else:
-                is_home_page = False
 
-            akamai_cache_reset(url_paths, invalidate_root=is_home_page, user_email=latest.user.email)
+
+def flush_akamai(page, is_live):
+    if is_live and settings.ENABLE_AKAMAI_CACHE_PURGE:
+        from publish_eccu.publish import publish as akamai_cache_reset
```
Comment (vccabral) [inline cfgov/v1/wagtail_hooks.py]: import on top?

File: cfgov/v1/wagtail_hooks.py [hunk 5]
```diff
@@ -33,37 +40,42 @@ def check_permissions(parent, user, is_publishing, is_sharing):
...
+
+        url_paths = [page.url_path.replace('cfgov/', '')]
```
Comment (vccabral) [inline cfgov/v1/wagtail_hooks.py]: Magic string. Why is this hardcoded?

Comment (kurtw) [inline cfgov/v1/wagtail_hooks.py]: Most of this code is leftover and I didn't bother refactoring it in my PR #2189

**Comment by anselmbradford** [comment]
LGTM RIP flapjack 😭

**Comment by rosskarchner** [comment]
👍

**Pull request merged**